src/subscription_manager/async.py

```diff
@@ -76,10 +78,7 @@ def __init__(self, certlib):
 
     def _run_bind(self, pool, quantity, bind_callback, cert_callback, except_callback):
         try:
-            self.plugin_manager.run("pre_subscribe", consumer_uuid=self.identity.uuid,
-                    pool_id=pool['id'], quantity=quantity)
-            ents = self.cp_provider.get_consumer_auth_cp().bindByEntitlementPool(self.identity.uuid, pool['id'], quantity)
-            self.plugin_manager.run("post_subscribe", consumer_uuid=self.identity.uuid, entitlement_data=ents)
+            attach.AttachService(self.cp_provider.get_consumer_auth_cp()).attach_pool(pool['id'], quantity)
```

Comment:
`attach.AttachService(self.cp_provider.get_consumer_auth_cp()).attach_pool(pool['id'], quantity)`

`.attach_pool` is the invocation.